dev-libs/smooth/smooth-0.8.74.0~pre4.recipe

```diff
@@ -0,0 +1,90 @@
+SUMMARY="A cross-platform C++ class library"
+DESCRIPTION="smooth is an object oriented C++ class library for Windows, \
+macOS, Linux and most Unix-like operating systems. It provides basic \
+functionality and platform support for applications and libraries.
+
+Features provided by smooth include:
+
+- user interface API with various widgets
+- simple to use multithreading API
+- file and network IO interface
+- completely transparent Unicode and software internationalization support
+- a libxml2 based XML parser"
+HOMEPAGE="http://www.smooth-project.org/"
+COPYRIGHT="1998-2018 by Robert Kausch"
+LICENSE="Artistic v2.0"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/smooth/files/smooth/0.8.74.0-pre4/smooth-0.8.74.0-pre4.tar.gz"
+CHECKSUM_SHA256="815f0329b5546c78feeb9ad1813855dfcb7bc062e28de44d0b4aadd7c6625d80"
+
+if [[ $portVersion == *~* ]]; then
+	SOURCE_DIR="$portName-${portVersion%%~*}-${portVersion##*~}"
+else
+	SOURCE_DIR="$portName-$portVersion"
+fi
+
+ARCHITECTURES="x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="${portVersion%~*}"
+libVersionCompat="$libVersion compat >= ${libVersion%.*}"
+
+PROVIDES="
+	smooth$secondaryArchSuffix = $portVersion
+	cmd:smooth_translator$secondaryArchSuffix = $portVersion
+	lib:libsmooth$secondaryArchSuffix = $libVersionCompat
+	lib:libsmooth_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
```

Comment:
```diff
- 	lib:libpng$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
```
(This is an arbitrary choice we made to make it easier to handle the transition from libpng `1.6` to `1.7` when this becomes available.)